lib/new_relic/agent/method_tracer_helpers.rb

```diff
@@ -68,10 +68,10 @@ def clm_enabled?
       end
 
       # The string representation of a singleton class looks like
-      # '#<Class:MyModule::MyClass>'. Return the 'MyModule::MyClass' part of
-      # that string
+      # '#<Class:MyModule::MyClass>', or '#<Class:MyModule::MyClass(id: integer, attrbiute: string)>'
+      # Return the 'MyModule::MyClass' part of that string
       def klass_name(object)
-        name = Regexp.last_match(1) if object.to_s =~ /^#<Class:(.*)>$/
+        name = Regexp.last_match(1) if object.to_s =~ /^#<Class:([\w:]*).*>$/
```

Comment:
I think we should require (`+`) the matched value instead of having it be optional (`*`):

```suggestion
        name = Regexp.last_match(1) if object.to_s =~ /^#<Class:([\w:]+).*>$/
```

(in retrospect, I also think the original pattern should have used `.+` instead of `.*`)